tests/functional/test_endpoints_v2.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2012-2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import json
+import os
+
+import pytest
+
+from botocore.endpoints_v2 import (
+    EndpointProvider,
+    EndpointResolutionError,
+    RuleSetEndpoint,
+)
+from botocore.loaders import Loader
+
+
+@pytest.fixture
+def partitions():
+    loader = Loader()
+    return loader.load_data('partitions')
+
+
+def ruleset_testcases(case_type):
+    loader = Loader()
+    services = loader.list_available_services('service-2')
+    test_data_dir = os.path.join(
+        os.path.dirname(__file__), 'data', 'endpoints'
+    )
+    services = loader.list_available_services('service-2')
+    for service in services:
+        try:
+            ruleset = loader.load_service_model(service, 'endpoint-rule-set')
+            with open(
+                os.path.join(test_data_dir, service, 'endpoint-tests.json')
+            ) as f:
+                data = f.read()
+                tests = json.loads(data)
+                for test in tests['testCases']:
+                    input_params = test['params']
+                    expected_object = test['expect']
+                    if case_type in expected_object:
```

Comment:
I think this is introducing risks that we won't actually run all the tests. What I was getting at earlier by creating a single static bucket is we evaluate the rules one time. Then with that data we partition everything into valid, error, and unknown. If we encounter any unknown cases we raise an exception and fail the test suite. 

Otherwise, we have two discrete collections that represent 100% of cases in the test files.